internal/app/edit/edit.go

```diff
@@ -1,79 +1,81 @@
 package edit
 
 import (
+	"bufio"
+	"bytes"
 	"fmt"
-	"os"
+	"path/filepath"
 	"slices"
-	"strings"
 
+	"github.com/DanWlker/remind/internal/data"
 	r_error "github.com/DanWlker/remind/internal/error"
-	"github.com/DanWlker/remind/internal/pkg/data"
-	"github.com/DanWlker/remind/internal/pkg/record"
-	"github.com/DanWlker/remind/internal/pkg/shared"
+	"github.com/DanWlker/remind/internal/record"
+	"github.com/DanWlker/remind/internal/shared"
 )
 
 func editTodoAssociatedWith(directory string) error {
-	recordItems, errGetRecordFileContents := record.GetFileContents()
-	if errGetRecordFileContents != nil {
-		return fmt.Errorf("helper.GetRecordFileContents: %w", errGetRecordFileContents)
+	recordItems, err := record.GetFileContents()
+	if err != nil {
+		return fmt.Errorf("helper.GetRecordFileContents: %w", err)
 	}
 
 	idx := slices.IndexFunc(recordItems, func(item record.RecordEntity) bool {
 		return item.Path == directory
 	})
 	if idx == -1 {
-		return &r_error.RecordDoesNotExistError{}
+		return r_error.RecordDoesNotExistError{ID: directory}
```

Comment:
This error type didn't need to be a pointer, but it did need its contents populated.

Reply:
Ahhh I was declaring the `Error()` function on the pointer(*) of `RecordDoesNotExistError` instead of the value no wonder I needed to return the pointer as the interface was on the pointer instead of the value